File: c/learn_struct.go
package main

import (
	"fmt"
)

type Person struct {
	name   string
	age    int
}


func main() {
	
	person1 := new(Person)
	person1.name = "My name is Slim.."
	person1.age = 45

	// var person2 *Person = person1
	//	person2 := person1
	person2 := person1
	
	var person3 = new(Person)
	person3.name = "Susin"
	person3.age = 30

	//var person4 **Person
	//person4 = &person3
	person4 := &person3
	//	fmt.Println("Person address:&p\n", (obj Person))
	fmt.Println("Person1 address:&p\n", &person1)
	fmt.Println("Person2 :%p\n", person2)
	fmt.Println("Person2 address:%p\n", &person2)
	
	fmt.Println(person1)
	fmt.Println("Now using pointers:\n", (*person2).name)

	fmt.Println("Person 3:\n", person3)
	fmt.Println("Person 3 address:\n", &person3)

	fmt.Println("Person 4 :\n", person4)
	fmt.Println("Person 4 address:\n", &person4)
	fmt.Println("Person 4 *p:\n", *person4)

}

File: c/learn_struct.c
#include <stdio.h>

struct Person  {
  char *name;
  int age;
  double z;
} person;


int main()
{
  struct Person *person1 = &person;
  struct Person *person2 = person1;
  
  person.name = "Susinthiran .S";
  person.age = 45;

  printf("Person sizeof: %d\n", sizeof(person));
  printf("Person address: %x\n", &person);
  printf("Person1 address: %x\n", &person1);
  printf("Person1 follow pointer: %x\n", (*person1));
  printf("Person2 address: %x\n", &person2);
  printf("Person2 follow pointer: %x\n", *person2);
  printf("Person2 name: %s\n", person2->name);

  printf("Person1 follow pointer: %x\n", (*person1));

  
  printf("Name: %s\n", person.name);
  printf("Age: %d\n", person.age);

  printf("Name through pointer: %s\n", person1->name);
  printf("Age through pointer: %d\n", person1->age);

  printf("Name through pointer#2: %s\n", (*person1).name);
  printf("Age through pointer#2: %d", (*person1).age);

}

  
